addons/binding/org.openhab.binding.opensprinkler/src/main/java/org/openhab/binding/opensprinkler/internal/api/OpenSprinklerHttpApiV100.java

```diff
@@ -0,0 +1,240 @@
+/**
+ * Copyright (c) 2014-2015 openHAB UG (haftungsbeschraenkt) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.opensprinkler.internal.api;
+
+import static org.openhab.binding.opensprinkler.internal.api.OpenSprinklerApiConstants.*;
+
+import org.openhab.binding.opensprinkler.internal.api.exception.CommunicationApiException;
+import org.openhab.binding.opensprinkler.internal.api.exception.GeneralApiException;
+import org.openhab.binding.opensprinkler.internal.util.Http;
+import org.openhab.binding.opensprinkler.internal.util.Parse;
+
+/**
+ * The {@link OpenSprinklerHttpApiV100} class is used for communicating with
+ * the OpenSprinkler API for firmware versions less than 2.1.0
+ *
+ * @author Chris Graham - Initial contribution
+ */
+public class OpenSprinklerHttpApiV100 implements OpenSprinklerApi {
+    protected final String hostname;
+    protected final int port;
+    protected final String password;
+
+    protected int firmwareVersion = -1;
+    protected int numberOfStations = DEFAULT_STATION_COUNT;
+
+    protected boolean connectionOpen = false;
+
+    /**
+     * Constructor for the OpenSprinkler API class to create a connection to the OpenSprinkler
+     * device for control and obtaining status info.
+     *
+     * @param hostname Hostname or IP address as a String of the OpenSprinkler device.
+     * @param port The port number the OpenSprinkler API is listening on.
+     * @param password Admin password for the OpenSprinkler device.
+     * @throws Exception
+     */
+    public OpenSprinklerHttpApiV100(final String hostname, final int port, final String password) throws Exception {
+        if (hostname == null)
+            throw new GeneralApiException("The given url is null.");
+        if (port < 1 || port > 65535)
+            throw new GeneralApiException("The given port is invalid.");
+        if (password == null)
+            throw new GeneralApiException("The given password is null.");
+        if (hostname.startsWith(HTTP_REQUEST_URL_PREFIX) || hostname.startsWith(HTTPS_REQUEST_URL_PREFIX))
+            throw new GeneralApiException("The given hostname does not need to start with " + HTTP_REQUEST_URL_PREFIX
+                    + " or " + HTTP_REQUEST_URL_PREFIX);
+
+        this.hostname = hostname;
+        this.port = port;
+        this.password = password;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public boolean isConnected() {
+        return connectionOpen;
+    }
+
+    /** {@inheritDoc} */
+    @Override
+    public void openConnection() throws Exception {
+        @SuppressWarnings("unused")
```

Comment:
you can remove returnContent and simply call Http.send.... in the try block. Same for the next occurances.